src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -16,6 +22,20 @@ export interface DataStructureFetchOptions {
   paginationToken?: string;
 }
 
+export interface DatasetIndexedView {
+  name: string;
+}
+
+export interface DatasetIndexedViewsService {
+  getIndexedViews: (dataset: Dataset) => Promise<DatasetIndexedView[]>;
+  /**
+   * Returns the data source saved object connected with the data connection object
+   * @param dataConnectionId Id of the data-connection saved object for which we want the data-source
+   * @returns Data source saved object for the Collection/Cluster attached with the data-connection
+   */
+  getConnectedDataSource: (dataConnectionId: string) => Promise<SavedObject>;
```

Comment:
nit:

```suggestion
  getDataSource: (dataset: Dataset) => Promise<SavedObject>;
```

if we were to go with this it might just be easier to just pass in the dataset as you have above instead of having it pass by id

Reply:
Don't want to name it `getDataSource` because it doesn't get the data source object for the given id but instead it looks for the data source saved object connected with the given data connection object